modules/apps/frontend-js/frontend-js-web/test/liferay/position/align.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {
+	Align,
+	align,
+} from '../../../src/main/resources/META-INF/resources/liferay/position/align';
+import {enterDocument, exitDocument} from './helpers';
+
+let center;
+let element;
+let mutable;
+let offsetParent;
+
+window.scrollTo = jest.fn();
+
+describe('Align', () => {
+	afterEach(() => {
+		exitDocument(center);
+		exitDocument(element);
+		exitDocument(mutable);
+		exitDocument(offsetParent);
+		window.scrollTo(0, 0);
+	});
+
+	beforeEach(() => {
+		enterDocument(
```

Comment:
I think this is basically:

```javascript
import {buildFragment} from '../../../build_fragment`; // (path not correct)

document.appendChild(buildFragment('<div... '));
```

Maybe we can avoid the `enterDocument` > `append` abstractions and helpers. Similar for `exitDocument` which should be something like:

```javascript
center && center.remove();
```

Reply:
Ok, that makes sense - I had a doubt when adding this in, but if we can avoid abstractions I think it'll be easier to read and maintain.

Reply:
Don't take this as avoid abstractions at all costs. As @wincent loves to say, ask yourself if the juice is worth the squeeze 🍋 !!

In this case, the key to me is that `enterDocument` has at least 3 branches (conditionals) of which you only ever use one path. That would be okay for a general-purpose library, but in your case it's adding unnecessary complexity for things that you don't even need.

Reply:
That's a very good way to reason about it - I'll keep that in mind (and on my PR checklist).